exe/genv-envs

```diff
@@ -17,17 +17,27 @@ QUERIES = {
     'config.gpu_memory': lambda env: env["config"].get("gpu_memory"),
 }
 
+
 def factory() -> Dict:
+    """
+    Create a 'empty' state object for genv environments.
+    :return: The state object (dict of dicts)
+    """
     return dict(
         envs={}
     )
 
+
 def cleanup(
         envs: Envs,
-        poll_pid: Callable[[int], bool]=genv.poll_pid,
-        poll_kernel: Callable[[str], bool]=genv.poll_kernel) -> None:
+        poll_pid: Callable[[int], bool] = genv.poll_pid,
+        poll_kernel: Callable[[str], bool] = genv.poll_jupyter_kernel) -> None:
     """
-    Clean up processes, kernels and environments.
+    Clean up processes, kernels and environments connected to the given environment.
+    :param envs: dictionary describing all existing environments.
+    :param poll_pid: a predicate which tells us which pids are still connected to the environment.
```

Comment:
Not exactly. Here I mean that the proccess are running inside the environment / effected directly by the environment. We need to specify that not 'all' running processes on the machine are effected, only these in the enviroment. How would you describe this relation?

Reply:
hmmm I see what you mean
you are right about "running" is not the good term here
but I guess that what "bothers" me is the term "_the_ environment" because we cleanup processes and kernels from _any_ environment is all

wdyt?

Reply:
I changed it to "an environment". This implies that we work on proccesses and kernels from any environments.